pkg/util/proxyutil/proxyutil.go

```diff
@@ -46,8 +46,22 @@ func ClearCookieHeader(req *http.Request, keepCookiesNames []string, skipCookies
 	keepCookies := map[string]*http.Cookie{}
 	for _, c := range req.Cookies() {
 		for _, v := range keepCookiesNames {
-			if c.Name == v {
+			// match all
+			if v == "*" {
```

Comment:
I believe `*` is a valid character for cookie names, according to: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#attributes so in theory this could "break" the current behavior.

Since we are only interested on this simplified scenario, I would introduce a new field in the JsonData `keepCookiePatterns` (right now, the field is  `keepCookies`) and just allow prefixes as patterns for the moment. The UX can be a bit more friendly than arbitrary adding `*`:

![image](https://github.com/grafana/grafana/assets/4025665/90142c74-3e25-4b20-b257-72b12593945e)

If you find any trouble with that, you may want to choose an invalid character like `[ ]` and hide that fact from the user. In the above mock, the value would be `keepCookies: "test[]"`.

Reply:
> I believe * is a valid character for cookie names, according to: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#attributes so in theory this could "break" the current behavior.

As discussed with @itsmylife we didn't consider this, but the general consensus of current suggested solution was perhaps "not the best UX, but at the same time forwarding of cookies is not very common use so lets start with the most simple thing we can do".

So I would personally risk breaking current behavior for not adding one additional configuration parameter for this use case. At least to start with. If this would cause problems we could potentially add a workaround by adding two stars, `**`, to denote it shouldn't match everything, rather the star.

Reply:
> If this would cause problems we could potentially add a workaround by adding two stars, **, to denote it shouldn't match everything, rather the star.

If we use `[]`, the solution would be as simple to implement and since those are not valid characters for cookies it should be safer. Any reason for not doing it?

Reply:
Guess not